p2p/protocol/ping/ping.go

```diff
@@ -33,7 +37,21 @@ func NewPingService(h host.Host) *PingService {
 }
 
 func (p *PingService) PingHandler(s network.Stream) {
-	buf := make([]byte, PingSize)
+	if err := s.Scope().SetService(ServiceName); err != nil {
+		log.Debugf("error attaching stream to ping service: %s", err)
+		s.Reset()
+		return
+	}
+
+	if err := s.Scope().ReserveMemory(PingSize, network.ReservationPriorityAlways); err != nil {
```

Comment:
Why not make priorities user configurable? Sounds like that's exactly what the user will want to do: tell the resource manager what's important for them.

Reply:
they are user configurable, it is a uint8 -- these are just some predefine levels.